src/java_bytecode/java_types.h

```diff
@@ -209,6 +213,20 @@ class java_generic_typet:public reference_typet
     return (generic_type_argumentst &)(
       add(ID_type_variables).get_sub());
   }
+
+  /// Replace every occurrence of a generic type argument with a given type.
+  /// \param argument generic type argument that should be replaced
+  /// \param type new type that the argument should be replaced with
+  void replace(const typet &argument, const typet &type)
```

Comment:
Outstanding without reply:
> Ah I know what confused me - method body inside the class definition, would prefer since not completely trivial to move into the source file.

Preference/non-blocking